### PR TITLE
Add option to choose where index or not the whole category tree of the product

### PR DESCRIPTION
**Pull request by JanPetr**
Implements #533 

Breaks hierachical widget of IS page. It's mentioned in comment of configuration input.

### CODE REVIEW

**Comment by maxiloc** [comment]
Is there a point to add this if it breaks the hierarchical faceting ? because in the end the issue was for the hierarchical widget. Using a refinementList with categories_without_path already works no ?

**Comment by JanPetr** [comment]
Yes, I got customer who didn't use hierachical widget and wanted filtering by categories. Because `categories_without_path` does not contain leveling and are without a path. So when you are in `Blazers /// Men` you would not be able to filter it as `categories_without_path` contains only "Blazers".

**Comment by maxiloc** [comment]
So for only for "Replace category page" ?

**Comment by JanPetr** [comment]
Yes, the issue is described in #533

**Comment by maxiloc** [comment]
Ok for me, we should add some kind of log message in case you put false + hierarchical category widget

File: app/design/frontend/base/default/template/algoliasearch/internals/configuration.phtml [hunk 1]
```diff
@@ -118,6 +118,17 @@ if ($config->isInstantEnabled()) {
...
+    /** @var Algolia_Algoliasearch_Helper_Logger $logger */
+    $logger = Mage::helper('algoliasearch/logger');
+    if ($logger->isEnable() && $config->indexWholeCategoryTree() === false) {
+        foreach ($config->getFacets() as $facet) {
+            if ($facet['attribute'] === 'categories') {
+                $logger->log('!! Warning !! You have enabled categories\' hierarchical widget but you do not index whole categories tree. The widget does not display correct data.');
+                break;
+            }
+        }
+    }
```
Comment (JanPetr) [inline app/design/frontend/base/default/template/algoliasearch/internals/configuration.phtml]: @maxiloc? OK?

Comment (maxiloc) [inline app/design/frontend/base/default/template/algoliasearch/internals/configuration.phtml]: yes perfect

**Pull request merged**